src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -447,6 +456,23 @@ public boolean isCredentialValid() throws IOException {
     }
 
     /**
+     * Ensures that the API URL is valid.
+     */
+    public boolean isApiUrlValid() throws IOException {
+        try {
+            GHApiInfo apiInfo = retrieve().to("/", GHApiInfo.class);
+            try {
+                new URL(apiInfo.getRateLimitUrl());
```

Comment:
Because these header are returned in any service of GitHub API. And the goal of this method is to verify if `apiUrl` is really a valid endpoint.